pkg/component/observability/monitoring/prometheus/component.go

```diff
@@ -339,6 +343,26 @@ func (p *prometheus) name() string {
 	return "prometheus-" + p.values.Name
 }
 
+func (p *prometheus) SetAdditionalAlertRelabelConfigs(configs []*monitoringv1.RelabelConfig) {
+	vals := make([]monitoringv1.RelabelConfig, len(configs))
+	for i, cfg := range configs {
+		vals[i] = *cfg
+	}
+	p.values.AdditionalAlertRelabelConfigs = vals
+}
+
+func (p *prometheus) GetAdditionalAlertRelabelConfigs() []*monitoringv1.RelabelConfig {
+	vals := p.values.AdditionalAlertRelabelConfigs
+	out := make([]*monitoringv1.RelabelConfig, len(vals))
+
+	for i := range vals {
+		cfg := vals[i]
+		out[i] = &cfg
+	}
```

Comment:
What about simplifying like this?
Otherwise, we're taking the memory address of `cfg`, which is being re-declared in each loop.
```suggestion
	for i := range vals {
		out[i] = &vals[i]
	}
```